source/particle/world.cc

```diff
@@ -54,10 +54,9 @@ namespace aspect
     {
       CitationInfo::add("particles");
       if (particle_load_balancing & ParticleLoadBalancing::repartition)
-#if DEAL_II_VERSION_GTE(9,4,0)
+
         this->get_triangulation().signals.weight.connect(
```

Comment:
Separately, remove the empty line you have here. It looks odd if to have
```
if (particle_load_balancing & ParticleLoadBalancing::repartition)

  this->get_triangulation().signals.weight.connect(
```
with an empty line in the middle.